wsagent/agent/src/main/resources/org.eclipse.che.ws-agent.script.sh

```diff
@@ -233,13 +242,20 @@ eval "DOWNLOAD_AGENT_BINARIES_URI=${DOWNLOAD_AGENT_BINARIES_URI}"
 
 if [ -f "${LOCAL_AGENT_BINARIES_URI}" ] && [ -s "${LOCAL_AGENT_BINARIES_URI}" ]
 then
-    AGENT_BINARIES_URI="file://${LOCAL_AGENT_BINARIES_URI}"
+    tar zxf "${LOCAL_AGENT_BINARIES_URI}" -C ${CHE_DIR}/ws-agent
 else
     echo "Workspace Agent will be downloaded from Workspace Master"
     AGENT_BINARIES_URI=${DOWNLOAD_AGENT_BINARIES_URI}
+    if [ ${CURL_INSTALLED} = true ]; then
+      curl -s  ${AGENT_BINARIES_URI} | tar  xzf - -C ${CHE_DIR}/ws-agent
+    else
+      # use wget
+      wget -qO- ${AGENT_BINARIES_URI} | tar xzf - -C ${CHE_DIR}/ws-agent
```

Comment:
Wget downloading section for terminal agent is more complex. Can you elaborate on why they differ and why we can't make them pretty the same?